R/intervals.r

```diff
@@ -205,7 +205,10 @@ interval <- function(start = NULL, end = NULL, tzone = tz(start)) {
   }
 
   if (length(start) == 0 || length(end) == 0) {
-    start <- POSIXct()
+    if (missing(tzone)) {
```

Comment:
`missing` is a bad programming practice + this logic is not correct. We want to force tzone and fail if it's called on wrong type.

Reply:
Okay, I think this is definitely a bug in some way though. If a `tzone` is specified but no `start` is specified, then an empty `UTC` POSIXct is used as the start, and the `tzone` is out of sync.

Open to suggestions!

``` r
library(lubridate)
#> 
#> Attaching package: 'lubridate'
#> The following object is masked from 'package:base':
#> 
#>     date

str(interval(tzone = "America/New_York"))
#> Formal class 'Interval' [package "lubridate"] with 3 slots
#>   ..@ .Data: num(0) 
#>   ..@ start: 'POSIXct' num(0) 
#>  - attr(*, "tzone")= chr "UTC"
#>   ..@ tzone: chr "America/New_York"
```

<sup>Created on 2020-03-22 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
Good catch. The logic there is a bit convoluted. I will have a look tomorrow.